pkg/ibmcsidriver/node.go

```diff
@@ -569,3 +569,15 @@ func (su *VolumeStatUtils) IsDevicePathNotExist(devicePath string) bool {
 	}
 	return false
 }
+
+func collectMountOptions(fsType string, mntFlags []string) []string {
+	var options []string
+	options = append(options, mntFlags...)
+
+	// By default, xfs does not allow mounting of two volumes with the same filesystem uuid.
+	// Force ignore this uuid to be able to mount volume + its clone / restored snapshot on the same node.
+	if fsType == "xfs" {
```

Comment:
Can we have some tests results for these changes?

Reply:
I had to extend NodeStage unit test with mount checks, it did not check if anything was mounted at all. Now it has a case for xfs + nouuid.

Reply:
@arahamad can you please take a look again?